package phy;

message fastSensingResult {
  required float rssi = 1;
}

message fineSensingResult {
  repeated float fft_bin = 1;
}

message demodStatistics {
  optional float rssi = 1; // received signal strength indicator [dB]
  optional float evm = 2;  // error vector magnitude [dB]
  optional float fer = 3;  // frame error rate
  optional float cfo = 4;  // carrier frequency offset (f/Fs)
}

message chStatusUpdate {
  required int32 channel = 1;
  optional string description = 2;

  enum statusType {
    FREE = 0;
    SU_BUSY = 1;
    PU_BUSY = 2;
  }

  required statusType currentStatus = 3;  
  optional string statusMessage = 4;
}

message channelDescription {
  required int32 f_center = 1;
  required int32 bandwidth = 2;
  optional int32 f_start = 3;
  optional int32 f_end = 4;
  optional string name = 5;
  optional int32 lo_offset = 6;
  optional string antenna = 7;
  required int32 tx_gain = 8;
  required int32 rx_gain = 9;
}
